flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/sink/WritableMetadata.java

```diff
@@ -83,4 +84,38 @@ public static TimestampMetadata of(List<String> metadataKeys, DataType physicalD
                     pos + physicalDataType.getLogicalType().getChildren().size());
         }
     }
+
+    /** Time-to-live metadata for HBase. */
+    public static class TimeToLiveMetadata extends WritableMetadata<Long> {
+
+        public static final String KEY = "ttl";
+        public static final DataType DATA_TYPE = DataTypes.BIGINT().nullable();
+
+        private final int pos;
+
+        public TimeToLiveMetadata(int pos) {
+            this.pos = pos;
+        }
+
+        @Override
+        public Long read(RowData row) {
+            if (pos < 0) {
+                return null;
+            }
+            if (row.isNullAt(pos)) {
+                throw new IllegalArgumentException(
+                        String.format("Writable metadata '%s' can not accept null value", KEY));
+            }
+            return row.getLong(pos);
+        }
+
+        public static TimeToLiveMetadata of(List<String> metadataKeys, DataType physicalDataType) {
```

Comment:
nit: I am not against static factory functions at all but in this particular case we could simply have one constructor that takes these 2 args, e.g.:
```java
public TimeToLiveMetadata(List<String> metadataKeys, DataType physicalDataType) {
    int idx = metadataKeys.indexOf(KEY);
    pos = idx < 0 ? -1 : idx + physicalDataType.getLogicalType().getChildren().size();
}
```
Because at the moment the `TimeToLiveMetadata(int)` ctor is only used in the static factory, so we could spare some code. The same is true for `TimestampMetadata`.

But feel free to ignore this comment if you prefer it this way.